plugins/out_s3/s3.c

```diff
@@ -252,6 +253,58 @@ static flb_sds_t concat_path(char *p1, char *p2)
     return dir;
 }
 
+static int read_seq_index(char *seq_index_file, uint64_t *seq_index,
+                          int *decrement_index)
+{
+    FILE *fp;
+    int ret;
+
+    fp = fopen(seq_index_file, "r");
+    if (fp == NULL) {
+        return -1;
+    }
+
+    ret = getw(fp);
+    if (ret == -1) {
+        return -1;
+    }
+    *seq_index = ret;
+
+    ret = getw(fp);
```

Comment:
So this worries me slightly because `getw` reads an amount of bytes from the file which is different on different systems. Its not platform independent.

But I think it is acceptable to use this function. I think for Fluent Bit we only care about 32 bit and 64 bit systems. On both, an int should be 4 bytes. 

Its important to always remember with C code- what assumptions am I making about what sort of system this code will run on?